proposals/collection-literals.md

```diff
@@ -112,23 +112,48 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
+  * If `Ei` is a *spread_element* with enumerated element type `Si` there is an implicit conversion from `Si` to `T`.
 
-* Single dimensional arrays (e.g. `T[]`)
-* [*Span types*](#span-types)
-* Types with a suitable [`Construct` method](#construct-methods)
-* Types that implement `System.Collections.IDictionary` - *dictionary [collection initializers](#collection-initializers)*
-* Interface types _`I<TKey, TValue>`_ implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
-* Types that implement `System.Collections.IEnumerable` - [*collection initializers*](#collection-initializers)
-* Interface types _`I<T>`_ implemented by `System.Collections.Generic.List<T>` (e.g. `IEnumerable<T>`, `IList<T>`, `IReadOnlyList<T>`)
+* To a *type* with an associated *[builder](#construct-methods)* where there is an implicit collection literal conversion from the collection literal to the *builder argument type*.
+
+* To a *type* that implements `System.Collections.IDictionary` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter `int capacity`.
+  * For each element `Ei` there is an applicable instance indexer for the *key_value_pair* from:
+    * If `Ei` is an *expression_element* with type `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from `Ei`.
+    * If `Ei` is a *dictionary_element* `Ki:Vi`, the values `Ki` and `Vi`.
+    * If `Ei` is a *spread_element* with enumerated element `Si` with type `dynamic` or `System.Collections.Generic.KeyValuePair<Ki, Vi>`, the values of applicable properties `Key` and `Value` from `Si`.
+
+* To an *interface type* *`I<K, V>`* where `I<TKey, TValue>` is implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` where for each element `Ei`:
+  * If `Ei` is an *expression_element* then the type of `Ei` is `dynamic`, or the type of `Ei` is a _type_ `System.Collections.Generic.KeyValuePair<Ki, Vi>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *dictionary_element* `Ki:Vi` there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * If `Ei` is a *spread_element* with enumerated element type `Si` then the type of `Si` is `dynamic`, or the type of `Si` is a _type_ `System.Collections.Generic.KeyValuePair<Ki, Vi>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+
+* To a *type* that implements `System.Collections.IEnumerable` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter `int capacity`.
```

Comment:
i assume thsi wording means we would work with constructrs with optional parameters.

Reply:
Yes, this wording was to allow optional parameters and trailing `params` parameter.